torch_xla/csrc/xla_graph_executor.cpp

```diff
@@ -1028,7 +1000,17 @@ XLAGraphExecutor::ScheduleSyncTensorsGraph(
     const std::vector<torch::lazy::BackendDataPtr>& tensor_data_vec) {
   auto tensors_data =
       SetTensorData(tensors, coll->config, coll->indices, tensor_data_vec);
```

Comment:
It seems like we have to override some of the work done in `SetTensorData` in `PrepareOutputShardingPropagation`, is there a reason we can't call `PrepareOutputShardingPropagation` first?

Reply:
Good question, and the answer is no (though we can make changes to both `SetTensorData` and `PrepareOutputShardingPropagation` to make it work in the reverse order, more work) sine we still want to hit `handle == nullptr` case there. I think since output sharding prop is conditional, I would prefer keeping the non SPMD code path intact if possible.